TTA overview widget

kryswisnaskas opened the pull request:
## Description of change
This PR builds on #300 adding support for a TTA overview widget on the landing page.


## How to test
- pull changes or run on sandbox
- create activity reports, both approved and not approved
- login with a user that has permissions to multiple regions
- (simulating logging with a different user can be done by running locally, stopping the server, changing the `CURRENT_USER_ID=<number>` in `.env` to a different number and then starting the server)
- note that having access to multiple regions allows to see the region select dropdown
- use the dropdown to change the region
- note that the overview widget, alerts and reports should adjust
- switch back to the previous region
- apply filters 
- switch back and forth between regions
- verify that the filters stay applied

Note * - outstanding issue - currently the dropdown menu won't close on blur

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-143


## Checklists

### Every PR

<!-- Add details to each completed item -->
- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

dcloud reviewed (commented):
It looks like the ticket is still in-progress, so I can leave some initial comments. I think this looks good so far. I can see that some styles may still need some work, such as the size of the widget numbers not being larger than the labels. Also, I'm looking forward to the llama! 

There was also discussion this morning about what this widget might look like at smaller screen sizes, though I don't know decisions were reached. If the widget needs to appear on iPad-like sizes (1024x768), I think it should use more screen real estate than the `grid-col-8` allows.

The region select will also need some size adjustments for smaller screens since right now the button's chevron overlaps the text, making the button hard to read. The functionality of the region select is nice, though it does cause some content to appear to flicker during the change. I wonder if we can improve the visuals to make the change smoother.

kryswisnaskas commented:
Thanks for the initial review. I fixed the styling, expanded column size, adjusted the sizing for smaller screens and also removed the coloring on the .svg checkmark in hopes that the flickering will disappear. I haven't been seeing flickers on my screen, so I it's helpful to know the display might not be smooth.
I am also looking forward to the llama. We ended up splitting that ticket into 2. llama will be coming under:
https://ocio-jira.acf.hhs.gov/browse/TTAHUB-161

thewatermethod commented:
To my eyes, the styling looks pretty good, all the font sizes and layout

I have a question about the widths of the columns on the overview widget. The first column uses the "grid-col-3" and looks kind of wide compared to the others, while the last column on the overview widget, Hours of TTA, uses "grid-col-1" and the type wraps to a second line. Not sure if this is intentional or not. 

I'd also note that the overview widget doesn't break down into additional rows as the screen shrinks, which would be my expected behavior. Again, this might be a known omission/non-issue.

I am also seeing some rapid layout reflow when switching from region to region. I think that you could probably stop some of that by reserving that layout space with CSS, or perhaps utilizing a "loading" state. It might be worth breaking that out into a separate bug on Jira as it's conceivable that this is a nice to have.

thewatermethod commented on frontend/src/pages/Landing/RegionalSelect.js:
```diff
@@ -0,0 +1,156 @@
+
+  // const delayedCloseMenu = () => setTimeout(setMenuIsOpen(false), 1000);
+
+  const CustomOption = (props1) => {
```
I'm a bit confused by "props1." Does the 1 signify something?

kryswisnaskas replied:
1 was used to distinguish between props. Ideally, we would want to move CustomOption out to avoid this. That was my plan.

kryswisnaskas replied:
I did run into problems though with the move. Instead, I renamed props1 to better describe and possibly refactor that code later if needed.

thewatermethod commented on frontend/src/pages/Landing/RegionalSelect.js:
```diff
@@ -0,0 +1,156 @@
+
+  const options = [...getUserOptions(regions), { custom: true }];
+  return (
+    <Select
```
Sorry to jump back in to flag another issue, but I am using this in what I'm working on and having trouble keyboard focusing this select box. I don't immediately see a prop you can pass to react-select to easily fix this, but I've only briefly looked and will continue to do so.

kryswisnaskas replied:
This wasn't my top priority for this PR mainly because of the limited audience. How about we address this separately, so that this PR goes in? That way they will at least have access to the data. Thanks for flagging it though.

kryswisnaskas commented:
> I'd also note that the overview widget doesn't break down into additional rows as the screen shrinks, which would be my expected behavior. Again, this might be a known omission/non-issue.

I run this by @arickalewis1 . She accepted it as is for now, but will go over the implementation once it's deployed and open another issue with outstanding tweaks to the UI.

kryswisnaskas commented:
> I am also seeing some rapid layout reflow when switching from region to region. I think that you could probably stop some of that by reserving that layout space with CSS, or perhaps utilizing a "loading" state. It might be worth breaking that out into a separate bug on Jira as it's conceivable that this is a nice to have.

I disabled the loading for this widget. Yes, we might enter this as a future enhancement.

kryswisnaskas commented:
> I have a question about the widths of the columns on the overview widget. The first column uses the "grid-col-3" and looks kind of wide compared to the others, while the last column on the overview widget, Hours of TTA, uses "grid-col-1" and the type wraps to a second line. Not sure if this is intentional or not.

Fixed.

thewatermethod reviewed (approved):
Looks great! I made a note to myself to write a JIRA issue for keyboard navigating the select box

kryswisnaskas commented:
The outstanding issues - keyboard navigation and closing on blur are documented here:

https://ocio-jira.acf.hhs.gov/browse/TTAHUB-197
https://ocio-jira.acf.hhs.gov/browse/TTAHUB-198
